src/full/Agda/Syntax/Concrete/Pattern.hs

```diff
@@ -103,8 +103,16 @@ lhsCoreApp core ps = core { lhsPats = lhsPats core ++ ps }
 
 -- | Add with-patterns to the right.
 lhsCoreWith :: LHSCore -> [Pattern] -> LHSCore
-lhsCoreWith (LHSWith core wps []) wps' = LHSWith core (wps ++ wps') []
-lhsCoreWith core                  wps' = LHSWith core wps' []
+lhsCoreWith (LHSWith core ell wps []) wps' = LHSWith core ell (wps ++ wps') []
+lhsCoreWith core                      wps' = LHSWith core False wps' []
+
+-- Andreas, 2020-06-04, issue #4704:
+-- The type-checker does not deal correctly with consecutive LHSWith,
+-- thus we fuse them.  (I did not get to the bottom of this problem.)
+-- | Smart 'LHSWith' constructor, fuses consecutive 'LHSWith'.
+lhsWith :: LHSCore -> Bool -> [Pattern] -> [NamedArg Pattern] -> LHSCore
+lhsWith (LHSWith core ell0 wps0 []) ell wps ps = LHSWith core (ell0 || ell) (wps0 ++ wps) ps
+lhsWith core                        ell wps ps = LHSWith core ell wps ps
```

Comment:
Are you sure this does not lose some information about the location of the `WithP`s? This information is used to reconstruct the ellipsis in pretty-printing.

Reply:
I am not sure about this function.  I thought it might fix the `Unexpected with patterns` problem, but it did not.  
But if you think about the original desugaring of ellipses, there would be a single `LHSWith` in `LHSCore` anyway (absent of #2822)---this invariant was what I tried to ensure here.